crest/Assets/Crest/Crest/Shaders/Resources/SdfJumpFlood.compute

```diff
@@ -28,14 +28,14 @@ void SdfJumpFlood(uint3 id : SV_DispatchThreadID)
 	bool downIsInBounds  = id.y >= _jumpSize;
 
 	float2 candidatePositions[8];
-	candidatePositions[0] = upIsInBounds                      ? _FromTexture[id.xy + uint2(0,           _jumpSize)] : currentNearestPoint;
-	candidatePositions[1] = upIsInBounds && rightIsInBounds   ? _FromTexture[id.xy + uint2(_jumpSize ,  _jumpSize)] : currentNearestPoint;
-	candidatePositions[2] = rightIsInBounds                   ? _FromTexture[id.xy + uint2(_jumpSize ,          0)] : currentNearestPoint;
-	candidatePositions[3] = rightIsInBounds && downIsInBounds ? _FromTexture[id.xy + uint2(_jumpSize , -_jumpSize)] : currentNearestPoint;
-	candidatePositions[4] = downIsInBounds                    ? _FromTexture[id.xy + uint2(0         , -_jumpSize)] : currentNearestPoint;
-	candidatePositions[5] = downIsInBounds && leftIsInBounds  ? _FromTexture[id.xy + uint2(-_jumpSize, -_jumpSize)] : currentNearestPoint;
-	candidatePositions[6] = leftIsInBounds                    ? _FromTexture[id.xy + uint2(-_jumpSize,          0)] : currentNearestPoint;
-	candidatePositions[7] = leftIsInBounds && upIsInBounds    ? _FromTexture[id.xy + uint2(-_jumpSize,  _jumpSize)] : currentNearestPoint;
+	candidatePositions[0] = upIsInBounds                      ? _FromTexture[id.xy + uint2(0,                       _jumpSize)] : currentNearestPoint;
+	candidatePositions[1] = upIsInBounds && rightIsInBounds   ? _FromTexture[id.xy + uint2(_jumpSize,               _jumpSize)] : currentNearestPoint;
+	candidatePositions[2] = rightIsInBounds                   ? _FromTexture[id.xy + uint2(_jumpSize,                       0)] : currentNearestPoint;
+	candidatePositions[3] = rightIsInBounds && downIsInBounds ? _FromTexture[id.xy + uint2(_jumpSize,        -(int) _jumpSize)] : currentNearestPoint;
```

Comment:
just realised - i'm not sure how this is working - casting a negative number to a uint - does that work due to some kind of wrapping behaviour? or is directx being very lenient (as usual..).

other platforms may be less forgiving. is it worth subtracting as int, then casting the result to uint. or making everything operate on ints? the cast may be a non-trivial operation on gpu (i'm not sure but it seems feasible).

Reply:
I will do more research in this. The hope is that two's compliment means that if you have unsigned wrapping everything should work fine. But will see if I can represent this more explicitly.